### PR TITLE
Provide the option to do single mutations or use the BufferedMutator

### PR DESCRIPTION
**Pull request by mgarolera**
Hopefully this will help troubleshoot issues that are obfuscated by the buffered mutator.

### CODE REVIEW

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 1]
```diff
@@ -760,6 +754,65 @@ public void finishBundle(Context context) throws Exception {
...
+    public void processElement(ProcessContext context) throws Exception {
+      incrementOperationCounter();
+
+      Mutation mutation = context.element();
+      if (DOFN_LOG.isTraceEnabled()) {
+        DOFN_LOG.trace("Persisting {}", Bytes.toStringBinary(mutation.getRow()));
+      }
+
+      if (mutation instanceof Put) {
+        getTable().put((Put)mutation);
```
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: I think there needs to be a space before mutation.

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 2]
```diff
@@ -760,6 +754,65 @@ public void finishBundle(Context context) throws Exception {
   }
 
   /**
+   * A {@link DoFn} that can write either a bounded or unbounded {@link PCollection} of
+   * {@link Mutation}s to a table specified via a {@link CloudBigtableTableConfiguration}.
+   */
+  public static class CloudBigtableSingleTableWriteFn
```
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: This should not be the default implementation.  Buffered mutator should be the default.

**Comment by mgarolera** [comment]
PTAL

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 3]
```diff
@@ -842,7 +898,14 @@ public PDone apply(PCollection<T> input) {
   public static PTransform<PCollection<Mutation>, PDone> writeToTable(
       CloudBigtableTableConfiguration config) {
     validate(config, config.getTableId());
-    return new CloudBigtableWriteTransform<>(new CloudBigtableSingleTableWriteFn(config));
+
```
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: Nit: Do you think the following is clearer to read?

DoFn<Mutation, Void> writeFn = null;
if (...) {
  writeFn = new CloudBigtableSingleTableSerialWriteFn(config);
} else {
 writeFn = CloudBigtableSingleTableBufferedWriteFn(config);
}
return new CloudBigtableWriteTransform<>(writeFn);

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 4]
```diff
@@ -760,6 +754,68 @@ public void finishBundle(Context context) throws Exception {
   }
 
   /**
+   * A {@link DoFn} that can write either a bounded or unbounded {@link PCollection} of
+   * {@link Mutation}s to a table specified via a {@link CloudBigtableTableConfiguration} doing a
+   * a single mutation per processElement call.
+   */
+  public static class CloudBigtableSingleTableSerialWriteFn
```
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: Can you add a test in CloudBigtableIOIntegrationTest for this?

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java [hunk 5]
```diff
@@ -682,22 +682,16 @@ public static Pipeline initializeForWrite(Pipeline p) {
...
-   * because the batch can fail half way, and Appends/Increments might be re-run causing the
-   * Mutation to be executed twice, which is never the user's intent. Re-running a Delete will not
-   * cause any differences. Re-running a Put isn't normally a problem, but might cause problems in
-   * some cases when the number of versions in the ColumnFamily is greater than one. In a case where
-   * multiple versions could be a problem, it's best to add a timestamp on the Put.
+   * {@link Mutation}s to a table specified via a {@link CloudBigtableTableConfiguration} using the
+   * BufferedMutator.
    */
-  public static class CloudBigtableSingleTableWriteFn
+  public static class CloudBigtableSingleTableBufferedWriteFn
```
Comment (sduskis) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java]: Please update CloudBigtableIOIntegrationTest.

**Comment by mgarolera** [comment]
Done. PTAL

**Comment by sduskis** [comment]
LGTM.  Please squash the commits first.

**Pull request merged**